test/colab/colab-test.mk

```diff
@@ -0,0 +1,31 @@
+.PHONY: colab-test-internal
+colab-test-internal: $(TEST_TOKEN_FILE) $(COLAB_TEST_HOME)/test_import_login_init.ipynb colab-test-pull
+	$(DOCKER_RUN)
+
+.PHONY: colab-test-internal-local
+colab-test-internal-local: $(TEST_TOKEN_FILE) $(COLAB_TEST_HOME)/test_import_login_init.ipynb colab-test-build
+	$(DOCKER_RUN)
+
+.PHONY: colab-test-pull
+colab-test-pull:
```

Comment:
Doesn't docker try to pull automatically if an image is not there?

Reply:
It does, but if the image is already there it won't pull again. This ensures that the latest image is always used. It also takes no time at all if the latest version is already present on the machine.

Reply:
In that case, you can just pass `--pull always` to `docker run`

Reply:
We have 2 usecases:
- Sometimes we want to build the image locally, run the image we just built locally, and also potentially push it from local to dockerhub
- Sometimes we just want to pull the latest from dockerhub and run that, e.g. in CI, or most devs on local as it's faster to pull and build from scratch.

If we update the run command with your suggestion, then for the first scenario we would build the image locally, and then pull the remote image and run that instead of the image built locally.

Reply:
Ok, thanks for explaining. This is a bit convoluted, but we can revisit when we do the automatic builds.